xla/backends/gpu/runtime/command_buffer_cmd.h

```diff
@@ -897,8 +899,9 @@ class WhileCmd : public CommandBufferCmd {
   std::optional<int64_t> trip_count_;
   bool enable_loop_unroll_ = false;
   bool is_unrolled_loop_ = false;
-  std::unique_ptr<se::CommandBuffer>
-      child_command_buffer_;  // The body command buffer for unrolled loop.
+  absl::Mutex mutex_;
```

Comment:
All state must be kept it `state` (StateManager passed as argument). It provides synchronization out of the box: http://google3/third_party/tensorflow/compiler/xla/backends/gpu/runtime/command_buffer_cmd.h;l=140-151;rcl=807940691, keeping any mutable state as a part of Cmd is big no-no.

```
state.GetOrCreate([] {
  return RecordChildCommands(...);
);
```